### ISSUE TITLE
Training error

### ISSUE THREAD

**Issue by meltingCat** [BindsNET/bindsnet#433]
Running `batch_eth_mnist.py` with batch_size 1, epoch 5, plot=False, gpu=True. It stops training at 256/60000 in every epoch.
```
Running on Device =  cuda

Begin training.

Progress: 0 / 5 (0.0000 seconds)
  0%|▌                                                                                                                                        | 256/60000 [00:31<2:00:48,  8.24it/s]
All activity accuracy: 8.20 (last), 8.20 (average), 8.20 (best)
Proportion weighting accuracy: 8.20 (last), 8.20 (average), 8.20 (best)

  0%|▌                                                                                                                                        | 256/60000 [00:32<2:04:44,  7.98it/s]
  0%|▌                                                                                                                                        | 256/60000 [00:31<1:59:34,  8.33it/s]
All activity accuracy: 29.30 (last), 18.75 (average), 29.30 (best)
Proportion weighting accuracy: 26.95 (last), 17.58 (average), 26.95 (best)

  0%|▌                                                                                                                                        | 256/60000 [00:31<2:01:20,  8.21it/s]
  0%|▌                                                                                                                                        | 256/60000 [00:31<2:00:41,  8.25it/s]
All activity accuracy: 40.23 (last), 25.91 (average), 40.23 (best)
Proportion weighting accuracy: 39.06 (last), 24.74 (average), 39.06 (best)

  0%|▌                                                                                                                                        | 256/60000 [00:31<2:01:43,  8.18it/s]
  0%|▌                                                                                                                                        | 256/60000 [00:30<1:58:22,  8.41it/s]
All activity accuracy: 29.69 (last), 26.86 (average), 40.23 (best)
Proportion weighting accuracy: 29.30 (last), 25.88 (average), 39.06 (best)

  0%|▌                                                                                                                                        | 256/60000 [00:30<2:00:06,  8.29it/s]
  0%|▌                                                                                                                                        | 256/60000 [00:31<2:00:13,  8.28it/s]
All activity accuracy: 36.33 (last), 28.75 (average), 40.23 (best)
Proportion weighting accuracy: 34.77 (last), 27.66 (average), 39.06 (best)

  0%|▌                                                                                                                                        | 256/60000 [00:31<2:01:58,  8.16it/s]
Progress: 5 / 5 (156.8054 seconds)
Training complete.


Begin testing

Test progress: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10000/10000 [15:54<00:00, 10.27it/s]
All activity accuracy: 0.49
Proportion weighting accuracy: 0.51 

Progress: 5 / 5 (954.1750 seconds)
Testing complete.

Test progress: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10000/10000 [15:54<00:00, 10.48it/s]
```

**Comment by Hananel-Hazan** [comment]
Thank you for bring this to out attention. 
We will look into it.

**Comment by ChatSam** [comment]
Working on this now

**Comment by Hananel-Hazan** [comment]
Fix with PR #437 
Thanks @ChatSam for the PR!
